libs/wingsdk/src/target-tf-azure/bucket.inflight.ts

```diff
@@ -0,0 +1,128 @@
+import { DefaultAzureCredential } from "@azure/identity";
+import { BlobServiceClient } from "@azure/storage-blob";
+import { BucketDeleteOptions, IBucketClient } from "../cloud";
+
+export class BucketClient implements IBucketClient {
+  private readonly bucketName: string;
+  private readonly blobServiceClient: BlobServiceClient;
+  private readonly defaultAzureCredential: DefaultAzureCredential =
+    new DefaultAzureCredential();
+
+  constructor(
+    bucketName: string,
+    storageAccount: string,
+    blobServiceClient?: BlobServiceClient
+  ) {
+    this.bucketName = bucketName;
+    this.blobServiceClient =
+      blobServiceClient ??
+      new BlobServiceClient(
+        `https://${storageAccount}.blob.core.windows.net`,
+        this.defaultAzureCredential
+      );
+  }
+
+  /**
+   * Put object into bucket with given body contents
+   *
+   * @param key Key of the object
+   * @param body string contents of the object
+   */
+  public async put(key: string, body: string): Promise<void> {
+    const containerClient = this.blobServiceClient.getContainerClient(
+      this.bucketName
+    );
+    await containerClient.getBlockBlobClient(key).upload(body, body.length);
+  }
+
+  /**
+   * Get object content from bucket
+   *
+   * @param key Key of the object
+   * @returns string contnet of the object as string
+   */
+  public async get(key: string): Promise<string> {
+    const containerClient = this.blobServiceClient.getContainerClient(
+      this.bucketName
+    );
+    const blobClient = containerClient.getBlobClient(key);
+
+    const downloadResponse = await blobClient.download();
+    if (downloadResponse.readableStreamBody === undefined) {
+      return "";
+    }
+
+    const downloaded = (
+      await this.streamToBuffer(downloadResponse.readableStreamBody)
+    ).toString();
+
+    return downloaded;
+  }
+
+  /**
+   * List all keys in the bucket
+   *
+   * @param prefix Limits the response to keys that begin with the specified prefix
+   * TODO - add pagination support
+   */
+  public async list(prefix?: string | undefined): Promise<string[]> {
```

Comment:
`prefix: string | undefined` and `prefix?: string` have pretty much the same effect, so no need to go double-optional
```suggestion
  public async list(prefix?: string): Promise<string[]> {
```